daphne/src/vdaf/mod.rs

```diff
@@ -606,7 +615,13 @@ impl VdafConfig {
         {
             // TODO spec: Consider removing the report ID from the AggregationJobResp.
             if helper.report_id != leader_report_id {
-                return Err(DapAbort::UnrecognizedMessage);
+                return Err(DapAbort::UnrecognizedMessage {
+                    detail: format!(
+                        "helper ({}) and leader report ID ({}) mismatch",
```

Comment:
I think we also need more context here: which reports are we comparing here? what error did the peer make that caused this to happen?

Reply:
Leaving this one unresolved, as I'm not sure my new message is correct. I added: `"The Leader is preparing report ID ({}), but it received {} from the Helper's last aggregate response. The Helper must return reports in the same order as AggregationJobInitReq.report_shares"`